core-plugins/src/test/java/co/cask/hydrator/plugin/batch/ETLDynamicPFSTestRun.java

```diff
@@ -0,0 +1,136 @@
+                                           BatchSource.PLUGIN_TYPE,
+                                           ImmutableMap.of(
+                                             "schema", recordSchema.toString(),
+                                             "name", "inputParquet",
+                                             "delay", "0d",
+                                             "duration", "1h"),
+                                           null);
+    ETLPlugin sinkConfig = new ETLPlugin("DynamicPFSParquet",
+                                         BatchSink.PLUGIN_TYPE,
+                                         ImmutableMap.of(
+                                           "schema", recordSchema.toString(),
+                                           "name", "outputParquet",
+                                           "fieldNames", "to_part_on"),
+                                         null);
+
+    ETLStage source = new ETLStage("source", sourceConfig);
+    ETLStage sink = new ETLStage("DynamicPFSParquet", sinkConfig);
+
+    ETLBatchConfig etlConfig = ETLBatchConfig.builder("* * * * *")
+      .addStage(source)
+      .addStage(sink)
+      .addConnection(source.getName(), sink.getName())
+      .build();
+
+    AppRequest<ETLBatchConfig> appRequest = new AppRequest<>(ETLBATCH_ARTIFACT, etlConfig);
+    Id.Application appId = Id.Application.from(Id.Namespace.DEFAULT, "parquetTest");
+    ApplicationManager appManager = deployApplication(appId, appRequest);
+
+    // write input to read
+    org.apache.avro.Schema avroSchema = new org.apache.avro.Schema.Parser().parse(recordSchema.toString());
+    List<GenericRecord> toWrite = new ArrayList<>();
+    toWrite.add(new GenericRecordBuilder(avroSchema)
+      .set("i", Integer.MAX_VALUE)
+      .set("to_part_on", "something_else")
+      .build());
+    toWrite.add(new GenericRecordBuilder(avroSchema)
+      .set("i", Integer.MAX_VALUE)
+      .set("to_part_on", "something")
+      .build());
+    DataSetManager<TimePartitionedFileSet> inputManager = getDataset("inputParquet");
+
+    long timeInMillis = System.currentTimeMillis();
+    inputManager.get().addPartition(timeInMillis, "directory");
+    inputManager.flush();
+    Location location = inputManager.get().getPartitionByTime(timeInMillis).getLocation();
+    location = location.append("file.parquet");
+    ParquetWriter<GenericRecord> parquetWriter =
+      new AvroParquetWriter<>(new Path(location.toURI()), avroSchema);
+    for (GenericRecord record : toWrite) {
+      parquetWriter.write(record);
+    }
+    parquetWriter.close();
+    inputManager.flush();
+
+    // run the pipeline
+    WorkflowManager workflowManager = appManager.getWorkflowManager(ETLWorkflow.NAME);
+    // add a minute to the end time to make sure the newly added partition is included in the run.
+    workflowManager.start(ImmutableMap.of("logical.start.time", String.valueOf(timeInMillis + 60 * 1000)));
```

Comment:
I'm not sure I follow.
I don't think we're using the workflow's logical start time for anything, are we?